addons/account/models/account_move.py

```diff
@@ -3090,6 +3091,7 @@ class AccountMoveLine(models.Model):
         help="This field is used for payable and receivable journal entries. You can put the limit date for the payment of this line.")
     currency_id = fields.Many2one('res.currency', string='Currency', required=True)
     partner_id = fields.Many2one('res.partner', string='Partner', ondelete='restrict')
+    partner_country_id = fields.Many2one(string="Partner's Country", related='partner_id.country_id', store=True)
```

Comment:
No need for `store=True` if we merge https://github.com/odoo/odoo/pull/66947

Reply:
indeed, we're waiting that task